### PR TITLE
Only pull out IDs from DB for /state_ids/ request

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

**Comment by NegativeMjark** [comment]
LGTM

File: synapse/handlers/federation.py [hunk 1]
```diff
@@ -1103,6 +1103,32 @@ def get_state_for_pdu(self, room_id, event_id):
...
+            _, state = state_groups.items().pop()
+            results = state
+
+            event = yield self.store.get_event(event_id)
+            if event and event.is_state():
+                # Get previous state
+                if "replaces_state" in event.unsigned:
+                    prev_id = event.unsigned["replaces_state"]
+                    if prev_id != event.event_id:
+                        results[(event.type, event.state_key)] = prev_id
```
Comment (NegativeMjark) [inline synapse/handlers/federation.py]: What's the point of this check?

File: synapse/handlers/federation.py [hunk 2]
```diff
@@ -1103,6 +1103,32 @@ def get_state_for_pdu(self, room_id, event_id):
             defer.returnValue([])
 
     @defer.inlineCallbacks
+    def get_state_ids_for_pdu(self, room_id, event_id):
```
Comment (NegativeMjark) [inline synapse/handlers/federation.py]: Maybe docstring to explain whether the state is before or after the event?

File: synapse/handlers/federation.py (hunk 1 above)
Comment (erikjohnston) [inline synapse/handlers/federation.py]: Because we don't want to put the event into the current state?

**Pull request merged**